client/extensions/woocommerce/state/data-layer/ui/shipping-zones/index.js

```diff
@@ -341,10 +346,55 @@ export const getCreateDefaultZoneActionListSteps = ( state ) => {
 	return steps;
 };
 
+const validateZone = ( state, dispatch, locationsFailAction, methodsFailAction ) => {
+	const siteId = getSelectedSiteId( state );
+	if ( ! areCurrentlyEditingShippingZoneLocationsValid( state, siteId ) ) {
```

Comment:
Sorry for not catching this earlier. You need to add a check for the `restOfTheWorld` zone, since in that case it **is** valid that it doesn't have any locations.